apps/antalmanac/src/stores/Schedules.ts

```diff
@@ -604,29 +608,38 @@ export class Schedules {
 
             // Get the course info for each course
             const courseInfoDict = new Map<string, { [sectionCode: string]: CourseInfo }>();
-            for (const [term, courseSet] of Object.entries(courseDict)) {
-                const sectionCodes = Array.from(courseSet);
-                // Code from ImportStudyList
-                const courseInfo = getCourseInfo(
-                    combineSOCObjects(
-                        await Promise.all(
-                            sectionCodes
-                                .reduce((result: string[][], item, index) => {
-                                    // WebSOC queries can have a maximum of 10 course codes in tandem
-                                    const chunkIndex = Math.floor(index / 10);
-                                    result[chunkIndex] ? result[chunkIndex].push(item) : (result[chunkIndex] = [item]);
-                                    return result;
-                                }, []) // https://stackoverflow.com/a/37826698
-                                .map((sectionCode: string[]) =>
-                                    queryWebsoc({
-                                        term: term,
-                                        sectionCodes: sectionCode.join(','),
-                                    })
-                                )
+            try {
+                for (const [term, courseSet] of Object.entries(courseDict)) {
+                    const sectionCodes = Array.from(courseSet);
+                    // Code from ImportStudyList
+                    const sectionsGroupedByTen = sectionCodes.reduce((accumulated, item, index) => {
+                        if (index % 10) {
+                            accumulated.push([item]);
+                        } else {
+                            accumulated.push([]);
+                            // The ? is to placate ts-lint. If accumulated is empty, the condition above would apply
+                            accumulated.at(-1)?.push(item);
+                        }
+                        return accumulated;
+                    }, [] as string[][]);
```

Comment:
Ok, I did some testing and I think this should work.
```suggestion
                    const sectionsGroupedByTen = sectionCodes.reduce((accumulated, item, index) => {
                        if (index % 10) {
                            accumulated.at(-1)?.push(item);
                        } else {
                            accumulated.push([item]);
                        }
                        return accumulated;
                    }, [] as string[][]);
```